packages/db/src/translate/granules.ts

```diff
@@ -1,11 +1,93 @@
 import Knex from 'knex';
 
-import { deconstructCollectionId } from '@cumulus/message/Collections';
+import { deconstructCollectionId, constructCollectionId } from '@cumulus/message/Collections';
+import { removeNilProperties } from '@cumulus/common/util';
 
 import { CollectionPgModel } from '../models/collection';
 import { PdrPgModel } from '../models/pdr';
-import { PostgresGranule } from '../types/granule';
+import { PostgresGranule, PostgresGranuleRecord } from '../types/granule';
 import { ProviderPgModel } from '../models/provider';
+import { FilePgModel } from '../models/file';
+import { translatePostgresFileToApiFile } from './file';
+import { getExecutionArnsByGranuleCumulusId } from '../lib/execution';
+
+/**
+ * Generate an API Granule object from a Postgres Granule with associated Files.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} granulePgRecord
+ *   Granule from Postgres
+ * @param {Knex | Knex.Transaction} knexOrTransaction
+ *   Knex client for reading from RDS database
+ * @param {Object} collectionPgModel - Instance of the collection database model
+ * @param {Object} pdrPgModel - Instance of the pdr database model
+ * @param {Object} providerPgModel - Instance of the provider database model
+ * @param {Object} filePgModel - Instance of the file database model
+ * @param {Object} executionPgModel - Instance of the execution database model
+ * @returns {Object} An API Granule with associated Files
+ */
+export const translatePostgresGranuleToApiGranule = async (
+  granulePgRecord: PostgresGranuleRecord,
+  knexOrTransaction: Knex | Knex.Transaction,
+  collectionPgModel = new CollectionPgModel(),
+  pdrPgModel = new PdrPgModel(),
+  providerPgModel = new ProviderPgModel(),
+  filePgModel = new FilePgModel()
+): Promise<AWS.DynamoDB.DocumentClient.AttributeMap> => {
```

Comment:
Shouldn't this typing match what's in @cumulus/types/api/ApirGranule?

Reply:
Yes, absolutely. 

https://github.com/nasa/cumulus/pull/2415/commits/a645406866c031b5c60ff835fb4b458ecfe1049a

This also highlighted a different in return types for the dates. They should be strings according to the `ApiGranule` type. Fixed in the above commit.

Reply:
Hmm, this is interesting.  We're not doing that for some of the other updatedAt fields, say executions.

Reply:
I'm wondering if this is a mistake.